src/ioctls/vcpu.rs

```diff
@@ -1049,6 +1049,31 @@ impl VcpuFd {
         Ok(reg_value)
     }
 
+    /// Notify the guest about the vCPU being paused.
+    ///
+    /// This signals to the host kernel that the specified guest is being paused by
+    /// userspace.  The host will set a flag in the pvclock structure that is checked
+    /// from the soft lockup watchdog.  The flag is part of the pvclock structure that
+    /// is shared between guest and host, specifically the second bit of the flags
+    /// field of the pvclock_vcpu_time_info structure.  It will be set exclusively by
+    /// the host and read/cleared exclusively by the guest.  The guest operation of
+    /// checking and clearing the flag must an atomic operation so
+    /// load-link/store-conditional, or equivalent must be used.  There are two cases
+    /// where the guest will clear the flag: when the soft lockup watchdog timer resets
+    /// itself or when a soft lockup is detected.  This ioctl can be called any time
+    /// after pausing the vcpu, but before it is resumed.
+    ///
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    pub fn kvmclock_ctrl(&self) -> Result<()> {
```

Comment:
Just curious, is it possible to add an `# Example` here and also some kind of test (at least in `test_faulty_vcpu_fd()` for example)?

Reply:
Yes I'll add those :)

Reply:
@lauralt I've updated `test_faulty_vcpu_fd()`, and added the example, but unfortunately for this ioctl to succeed, we need the VM to be running with the vCPU being paused at some point, which makes the whole setup a bit complex. That's why in the example, the call to `kvmclock_ctrl()` does not invoke `unwrap()` as it is failing otherwise.

Reply:
:( The way the example looks now, I'm not so sure it's useful as it doesn't show some minimal use case scenario. I would either call `unwrap_err()` instead and explain in a comment why this call fails and eventually what conditions should be met for it to work or drop the whole example. What do you think?

Reply:
I agree. That's why I'd drop the whole example as it does not provide any added value, which is supposed to be the point of these examples usually.